docs/planning/otel/specs/OTelWebSdk-Architecture.md

```diff
@@ -0,0 +1,2068 @@
+- **Configuration Inheritance**: Team-specific configuration overrides with enterprise-wide default policies
+- **Coordinated Management**: Central management capabilities for monitoring and controlling team instances
+
+```typescript
+interface IEnterpriseManager {
+  createTeamInstance(teamId: string, config?: Partial<IOTelWebSdkConfig>): IOTelWebSdk;
+  getTeamInstance(teamId: string): IOTelWebSdk | undefined;
+  getAllTeamInstances(): Map<string, IOTelWebSdk>;
+  setEnterprisePolicy(policy: IEnterprisePolicy): void;
+  unloadTeamInstance(teamId: string): Promise<void>;
+  unloadAllTeams(): Promise<IEnterpriseUnloadResult>;
+}
+
+interface IEnterprisePolicy {
+  defaultSamplingRate: number;
+  allowedConnectionStrings: string[];
+  requiredTelemetryProcessors: string[];
+  dataRetentionPolicy: IDataRetentionPolicy;
+  complianceSettings: IComplianceSettings;
+}
+
+// Example enterprise usage
+const enterpriseManager = createEnterpriseManager(enterpriseConfig);
+
+// Team A - E-commerce
+const ecommerceSDK = enterpriseManager.createTeamInstance('ecommerce', {
+  customProperties: { 'team.name': 'ecommerce', 'cost.center': 'retail' }
+});
+
+// Team B - Analytics
+const analyticsSDK = enterpriseManager.createTeamInstance('analytics', {
+  customProperties: { 'team.name': 'analytics', 'cost.center': 'insights' }
+});
+```
+
+### 10. Bundle Size Optimization
+- **Tree-Shaking Friendly**: Full support for dead code elimination with no side effects
+- **Modular Loading**: Individual components can be imported to minimize bundle size
+- **Code Splitting**: Automatic code splitting for large applications with dynamic imports
+- **Compression Optimization**: Optimal minification and compression for production builds
+- **Size Monitoring**: Built-in bundle size analysis and optimization recommendations
+
+```typescript
+// Selective imports for minimal bundle size
+import { createTraceProvider } from '@microsoft/otel-web-sdk/trace';
+import { createOTLPExporter } from '@microsoft/otel-web-sdk/exporters/otlp';
+
+// Or full SDK import with tree-shaking
+import { createOTelWebSdk } from '@microsoft/otel-web-sdk';
+
+// Bundle size targets:
+// Core SDK: ~15-20KB gzipped
+// With all providers: ~25-35KB gzipped  
+// With auto-instrumentation: ~40-55KB gzipped
+// Full feature set: ~60-80KB gzipped (still smaller than most alternatives)
+```
+
+### 11. Universal JavaScript Runtime Support
+- **Platform Detection**: Automatic detection of runtime environment (Browser, Web Worker, Node.js, SSR)
+- **API Adaptation**: Runtime-specific API usage with graceful fallbacks for missing features
+- **SSR Compatibility**: Full support for Server-Side Rendering scenarios with hydration
+- **Web Worker Support**: Complete functionality in Web Worker and Service Worker environments
+- **Node.js Integration**: Seamless operation in Node.js environments for SSR and build tools
+
+```typescript
```

Comment:
These examples are OVERKILL and this is how we don't want to implement this. We don't need to "know" upfront what features do and don't exist. At the point of attempting to "use" a feature we detect there and then (via helper functions) and if the API / feature is not present we just don't initialize that part of the code or we provide some gracefull fallback.